src/Aspire.Dashboard/Model/ResourceEndpointHelpers.cs

```diff
@@ -24,13 +24,21 @@ public static List<DisplayedEndpoint> GetEndpoints(ResourceViewModel resource, b
                     continue;
                 }
 
+                if (!includeNonEndpointUrls && string.IsNullOrEmpty(url.EndpointName))
+                {
+                    continue;
+                }
+
                 endpoints.Add(new DisplayedEndpoint
                 {
-                    Name = url.Name,
-                    Text = url.Url.OriginalString,
+                    Name = url.EndpointName ?? "",
```

Comment:
Why does this default to empty string instead of null?

Reply:
Being updated to default to `"-"` when there's no `EndpointName`.